src/ngx_http_lua_kong_request.c

```diff
@@ -0,0 +1,109 @@
+#include <ngx_config.h>
+#include <ngx_core.h>
+#include <ngx_http.h>
+
+/* name should be lowercased and preprocessed for both search functions */
+
+/* by hash */
+static ngx_str_t *search_known_header(ngx_http_request_t *r, ngx_str_t name)
+{
+    /* Calculate a hash of lowercased header name */
+    ngx_uint_t hash = 0;
+    for (size_t i = 0; i < name.len; i++) {
+        hash = ngx_hash(hash, name.data[i]);
+    }
+
+    ngx_http_core_main_conf_t *cmcf =
+        ngx_http_get_module_main_conf(r, ngx_http_core_module);
+    ngx_http_header_t *hh =
+        ngx_hash_find(&cmcf->headers_in_hash, hash, name.data, name.len);
+
+    /* There header is unknown or is not hashed yet. */
+    if (hh == NULL) {
+        return NULL;
```

Comment:
We can add some `ngx_log_debug1(NGX_LOG_DEBUG_HTTP, ...)` , then check the log in the test case.

Reply:
Sure. But better from another place.